crates/client/rpc/src/lib.rs

```diff
@@ -220,9 +226,101 @@ where
+                let contract_class = self
+                    .overrides
+                    .for_block_hash(self.client.as_ref(), current_block_hash)
+                    .contract_class_by_class_hash(current_block_hash, class_hash);
+
+                if let Some(contract_class) = contract_class {
+                    error!("Contract class already exists: {:?}", contract_class);
+                    // return Err(StarknetRpcApiError::ClassAlreadyDeclared.into());
+                }
+
+                let extrinsic =
+                    self.convert_tx_to_extrinsic(best_block_hash, AccountTransaction::Declare(transaction)).unwrap();
+
+                submit_extrinsic(self.pool.clone(), best_block_hash, extrinsic).await.ok()?;
+
+                Some((tx_hash, class_hash))
+            }
+            DeclareTransactionCommonInput::V0(declare_txn, program_vec, entrypoints, abi_length) => {
+                let txn_hash: TransactionHash = TransactionHash(StarkHash { 0: FieldElement::ONE.to_bytes_be() });
+
+                let program_decoded = Program::decode_all(&mut &*program_vec).unwrap();
+
+                let class_info = ClassInfo::new(
+                    &blockifier::execution::contract_class::ContractClass::V0(ContractClassV0(Arc::from(
+                        ContractClassV0Inner {
+                            program: program_decoded,
+                            entry_points_by_type: entrypoints,
+                        },
+                    ))),
+                    0,
+                    abi_length,
+                )
+                .unwrap();
+
+                let declare_transaction = DeclareTransaction::new(
+                    starknet_api::transaction::DeclareTransaction::V0(declare_txn),
+                    txn_hash,
+                    class_info,
+                )
+                .unwrap();
+
+                let current_block_hash = self.get_best_block_hash();
+                let contract_class = self
+                    .overrides
+                    .for_block_hash(self.client.as_ref(), current_block_hash)
+                    .contract_class_by_class_hash(current_block_hash, declare_transaction.class_hash());
+
+                if let Some(contract_class) = contract_class {
+                    error!("Contract class already exists: {:?}", contract_class);
+                    // return Err(StarknetRpcApiError::ClassAlreadyDeclared.into());
+                }
+
+                let extrinsic = self
+                    .convert_tx_to_extrinsic(best_block_hash, AccountTransaction::Declare(declare_transaction.clone()))
+                    .unwrap();
+
+                let res = submit_extrinsic(self.pool.clone(), best_block_hash, extrinsic).await.unwrap();
+
+                log::info!(">>>>> extrinsic res : {:?}", res);
+                log::info!(">>>>> class hash declared: {:?} ✅ | txn hash : {:?}", declare_transaction.class_hash(), declare_transaction.tx_hash);
```

Comment:
we can remove these logs. if you really want it, we can maybe make it a trace log.

Reply:
also, I just noticed we don't add a log target to any logs in the client like we do here - https://github.com/apoorvsadana/madara/blob/main/crates/pallets/starknet/src/lib.rs#L101-L101

can you create a new issue to do this for madara logs in client as well, thanks!

Reply:
still pending